newIDE/app/src/Utils/memoize.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+
+export default function<Input, Output>(func: (Input) => Output): (Input) => Output {
+  const primitives = new Map();
+  const objects = new WeakMap();
+
+  function cacheFor<Input, Output>(input: Input): Output {
+    const isObject: boolean = typeof input === 'object';
```

Comment:
No need to type things that can be trivially inferred by Flow:
```suggestion
    const isObject = typeof input === 'object';
```